docs/api.adoc

```diff
@@ -341,6 +341,136 @@ Signs transaction offline with private key from keystore file with given passphr
+     "id": 1}
+<-- {"jsonrpc": "2.0", "result": true, "id": 1}
+----
+
+=== emerald_listContracts
+
+Return the list of all not hidden contracts from the contract storage directory.
+
+*Parameters*:
+
+    * `additional` (Object, optional)
+    ** `chain` (String, optional) - chain name, by default `mainnet`, other possible variant `testnet`
+    ** `chain_id` (Number, optional) - chain id number, by default for `mainnet` it equals `61`
+    ** `show_hidden` (Boolean, optional) - also show hidden accounts
+
+*Result*:
+
+    * `contracts` (Array)
+    ** `contract` (Object) - a contract object
+    *** `name` (String, optional) - account name
+    *** `abi` (Array) - JSON format for a contract ABI
+    *** `address` (String) - hex-encoded 20 bytes contract address
+
+*Examples*:
+
+----
+--> {"jsonrpc": "2.0", "method": "emerald_listContracts", "params": [{"chain": "testnet", "show_hidden": true}], "id": 1}
+<-- {"jsonrpc": "2.0", "result":
+      [
+       {"name": "BitEther",
+        "address": "0x085fb4f24031eaedbc2b611aa528f22343eb52db",
+        "abi": [{"inputs":[],"name":"name","outputs":[{"name":"","type":"string"}]},{"inputs":[{"name":"_spender","type":"address"},{"name":"_value","type":"uint256"}],"name":"approve","outputs":[{"name":"success","type":"bool"}]},{"inputs":[{"name":"from","type":"address"},{"name":"to","type":"address"},{"name":"value","type":"uint256"}],"name":"Transfer"}]
+        }
+      ],
+     "id": 1}
+----
+
+=== emerald_encodeFunctionCall
```

Comment:
`call` name is used everywhere to refer to read something locally from blockchain. It can be confusing in the future, for me better to expend already existing `emerald_signTransaction` method to add support also for smart contract transactions.